mlir/test/python/ir/context_lifecycle.py

```diff
@@ -47,3 +47,17 @@
 assert '"mlir.ir.Context._CAPIPtr"' in repr(c4_capsule)
 c5 = mlir.ir.Context._CAPICreate(c4_capsule)
 assert c4 is c5
+c4 = None
+c5 = None
+gc.collect()
+
+# Create a global threadpool and use it in two contexts
+tp = mlir.ir.ThreadPool()
+assert tp.get_max_concurrency() > 0
+c5 = mlir.ir.Context()
+c5.enable_multithreading(False)
```

Comment:
Is this needed? What happens if you don't set it?

Reply:
That's what I feared, then we need to defend against it in the Python API. That is we shouldn't crash and instead either throw an exception when the user didn't disable multi-threading first, or do it for the user implicitly in the call to `set_thread_pool()`

Reply:
We can also take it as optional at construction indeed!

But can't there be cases where you handed over an existing context and this "set_thread_pool" API can be also useful?

Reply:
I'm just wondering if disabling multithreading and changing thread pools are safe things to do. Maybe it's fine.

Reply:
We can also call internally equivalent of `enable_multithreading(False)` inside `set_thread_pool()` in the binding?